### DISCUSSION TITLE
Accessing to a tuple first element as _id for a ref in schema

### DISCUSSION THREAD

**Discussion by idaromnet** [Automattic/mongoose#14369] [Q&A]
Hello everyone,
I have a Schema that has the following structure:
```
const mySchema = new mongoose.Schema(
  {
    field: {
      type: [Number], // note it is a tuple and I want basically mongoose only look at the first element of the tuple when it needs to populate this field with ref 
      required: true,
      ref:'Model'
      }
   })
```
So basically when I want to populate this field I am looking for a way that mongoose look at the first element of the array to detect what is the foreign key (_id) for my other model's document.
Anyone has any idea how could I achieve it?
Thanks

**Comment by vkarpov15** [comment]
As written, if you populate `field`, Mongoose will replace every element in the `field` array with the corresponding document with that id. Which would work, since it would get the first element. It will just try to fetch additional documents for the other elements in `field`. Does that help?
